hw/ip/otbn/doc/_index.md

```diff
@@ -0,0 +1,2514 @@
+On the other hand, the control flow is clearly separated from the data, and reduced to a minimum to avoid data leakage.
+
+The data OTBN processes is security-sensitive, and the processor design centers around that.
+The design is kept as simple as possible to reduce the attack surface and aid verification and testing.
+For example, no interrupts or exceptions are included in the design, and all instructions are designed to be executable within a single cycle.
+
+OTBN is designed as a self-contained co-processor with its own instruction and data memory, which is accessible as bus device.
+
+## Compatibility
+
+OTBN is not designed to be compatible with other cryptographic accelerators.
+Its design is inspired by dcrypto, the cryptographic accelerator used in Google’s Titan chips, which itself is inspired by the [Fiat Crypto Machine](http://adam.chlipala.net/papers/FiatCryptoSP19/FiatCryptoSP19.pdf).
+
+# Instruction Set
+
+OTBN is a processor with a custom instruction set, which is described in this section.
+The instruction set is split into two groups:
+
+* The **base instruction subset** operates on the 32b General Purpose Registers (GPRs).
+  Its instructions are used for the control flow of a OTBN application.
+  The base instructions are inspired by RISC-V’s RV32I instruction set, but not compatible with it.
+* The **big number instruction subset** operates on 256b Wide Data Registers (WDRs).
+  Its instructions are used for data processing.
+  Also included are compare and select instructions to perform data-dependent control flow in a safe fashion.
+
+The subsequent sections describe first the processor state, followed by a description of the base and big number instruction subsets.
+
+## Processor State
+
+### General Purpose Registers
+
+OTBN has 32 General Purpose Registers (GPRs).
+Each GPR is 32b wide.
+General Purpose Registers in OTBN are mainly used for control flow.
+The GPRs are defined in line with RV32I.
+
+<table>
+  <tr>
+    <td><code>x0</code></td>
+    <td><strong>Zero</strong>. Always reads 0. Writes are ignored.</td>
+  </tr>
+  <tr>
+    <td><code>x1</code></td>
+    <td>
+      <strong>Return address</strong>.
+      Access to the call stack.
+      Reading <code>x1</code> pops an address from the call stack.
+      Writing <code>x1</code> pushes a return address to the call stack.
+      Reading from an empty call stack results in an alert.
+    </td>
+  </tr>
+  <tr>
+    <td>...</td>
+    <td></td>
+  </tr>
+  <tr>
+    <td><code>x31</code></td>
+    <td><strong>General Purpose Register 31</strong>.</td>
+  </tr>
+</table>
+
+### Control and Status Registers (CSRs)
+
+Control and status registers are 32b wide.
+
+<table>
+  <thead>
+    <tr>
+      <th>Number</th>
+      <th>Privilege</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>0x7C0</td>
```

Comment:
are these `0x7c0` et al expected to be separated by 0x004 since they're all 32b wide? And I assume these `0x7c0` et al are not related to `x0 .. x31` correct? That table doesn't have a header row, so doesn't indicate `name` vs. this `number`, might be worth a header on that table.

Reply:
No relation to the GPRs. The table should have a header row (and has one in my rendering), let me know if your rendered version looks different.

![image](https://user-images.githubusercontent.com/1467123/84263462-71333400-ab17-11ea-9bd4-85aa31b0b5c0.png)

Reply:
In the interest of clarity, should we put an explicit comment that there is no relationship between GPR and CSR?  (and perhaps between GPR and WDR too?)